stonesoup/types/array.py

```diff
@@ -46,12 +77,24 @@ def __new__(cls, *args, **kwargs):
         return array.view(cls)
 
     def __array_wrap__(self, array):
-        return np.asarray(array)
+        return StateVector._cast(np.asarray(array))
 
     def __matmul__(self, other):
-        out = np.matmul(self, np.asfarray(other))
+        out = np.matmul(np.asfarray(self), np.asfarray(other))
         return out.view(type=type(self))
 
     def __rmatmul__(self, other):
-        out = np.matmul(np.asfarray(other), self)
+        out = np.matmul(np.asfarray(other), np.asfarray(self))
         return out.view(type=type(other))
+
+    def __sub__(self, other):
+        return type(self)(super().__sub__(other))
```

Comment:
Not really - the code was written a two different times. Using type(self) is more general approach since the code can be copied to other classes without modification. This was coded to get something working for the Beta release and fix the issues that cropped up with the unscented transform.

I prefer the approach I used in #104 - I was able to delve deeper into the Numpy support functions and find a better solution.